packages/web/src/common/store/upload/sagas.js

```diff
@@ -999,6 +1004,10 @@ function* uploadSingleTrack(track) {
           throw new Error(`Could not confirm upload single track ${trackId}`)
         }
 
+        // getting track immediately after acdc confirms
+        // can 404 because it's not indexed yet
+        yield delay(1500)
```

Comment:
Think I'd prefer to have a polling loop instead of the simple wait to give us a chance at hitting the first time. We can do something like
```
while (!response)
	api.getTrack
	if (!response) sleep
```
Not sure what the timing looks like on the indexing side, but this would give us good behavior whether it's fast or not.

If the change to reselection logic you proposed will get us the same behavior, I'm happy with that as well.

Reply:
+1 to the "polling" - except this misses the fact that the endpoint gives a 400 and we re-select discovery nodes, which we shouldn't do. So we need both of these changes basically.
I don't feel good merging the 1500ms delay because it'll still be non-deterministic. If the node you're talking to get stuck or takes longer than 1s to index the block, we're going to be in trouble still

And eventually POST endpoints to encapsulate all of this logic with optimistic writes :)